libnest2d_js/libnest2d_js.cpp

```diff
@@ -0,0 +1,247 @@
+
+    // Point class - fix the getter/setter type issue
+    class_<Point>("Point")
+        .constructor<>()
+        .constructor<long, long>()
+        .function("x", optional_override([](const Point& self) -> long { return getX(self); }))
+        .function("y", optional_override([](const Point& self) -> long { return getY(self); }))
+        .function("setX", optional_override([](Point& self, long value) { setX(self, value); }))
+        .function("setY", optional_override([](Point& self, long value) { setY(self, value); }))
+        ;
+
+    // Box class
+    class_<Box>("Box")
+        .constructor<Point>()
+        .constructor<Point, Point>()
+        .constructor<long, long, Point>()
+        .class_function("infinite", &Box::infinite)
+        .class_function("fromDimensions", optional_override([](long width, long height) -> Box {
+            return Box(width, height);
+        }))
+        .function("minCorner", select_overload<const Point&() const>(&Box::minCorner))
+        .function("maxCorner", select_overload<const Point&() const>(&Box::maxCorner))
+        .function("width", &Box::width)
+        .function("height", &Box::height)
+        .function("area", &Box::area<double>)
+        .function("center", select_overload<Point() const>(&Box::center))
+        ;
+
+    // Circle class
+    class_<Circle>("Circle")
+        .constructor<>()
+        .constructor<Point, double>()
+        .function("center", select_overload<const Point&() const>(&Circle::center))
+        .function("setCenter", select_overload<void(const Point&)>(&Circle::center))
+        .function("radius", select_overload<double() const>(&Circle::radius))
+        .function("setRadius", select_overload<void(double)>(&Circle::radius))
+        .function("area", &Circle::area)
+        ;
+
+    // NfpConfig::Alignment enum
+    enum_<NfpConfig::Alignment>("Alignment")
+        .value("CENTER", NfpConfig::Alignment::CENTER)
+        .value("BOTTOM_LEFT", NfpConfig::Alignment::BOTTOM_LEFT)
+        .value("BOTTOM_RIGHT", NfpConfig::Alignment::BOTTOM_RIGHT)
+        .value("TOP_LEFT", NfpConfig::Alignment::TOP_LEFT)
+        .value("TOP_RIGHT", NfpConfig::Alignment::TOP_RIGHT)
+        .value("DONT_ALIGN", NfpConfig::Alignment::DONT_ALIGN)
+        ;
+
+    // NfpConfig class
+    emscripten::value_object<NfpConfig>("NfpConfig")
+        .field("alignment", &NfpConfig::alignment)
+        .field("starting_point", &NfpConfig::starting_point)
+        .field("accuracy", &NfpConfig::accuracy)
+        .field("explore_holes", &NfpConfig::explore_holes)
+        .field("parallel", &NfpConfig::parallel)
+        ;
+
+
+    // Item class
+    class_<Item>("Item")
+        .constructor<const PolygonImpl&>()
+        .class_function("createFromVertices", optional_override([](const emscripten::val& jsVertices) -> Item {
+            std::vector<Point> vertices = jsArrayToPointVector(jsVertices);
```

Comment:
If we don't have to apply the conversion between `Point` and the js point val then we can simplify this code to, which again saves copying vertex data
```suggestion
            const auto vertices = emscripten::vecFromJSArray<Point>(jsVertices)
```